src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -404,7 +419,8 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		return new Promise<void>((resolve, reject) => {
 			this.updateModelServerCapabilities(model);
 			// If connecting from a query editor set "save connection" to false
-			if (params && params.input && params.connectionType === ConnectionType.editor) {
+			if (params && params.input && params.connectionType === ConnectionType.editor ||
```

Comment:
Change to this to avoid potential nullref if params in undefined. 
```TypeScript
if (params && params.input && (params.connectionType === ConnectionType.editor ||
				params.connectionType === ConnectionType.extension)) {
```